OpenRA.Game/Traits/World/ActorMap.cs

```diff
@@ -128,6 +132,7 @@ public void Tick(ActorMap am)
 				var delta = new WVec(range, range, WDist.Zero);
 				currentActors = am.ActorsInBox(position - delta, position + delta)
 					.Where(a => (a.CenterPosition - position).HorizontalLengthSquared < range.LengthSquared)
+					.Where(a => (vRange.Length < 1) || Math.Abs(a.CenterPosition.Z - position.Z) <= vRange.Length)
```

Comment:
Minor quibbling follows:

Brackets around `vRange.Length < 1` are not needed.

Also,  you could instead do (but you don't have to):

```c#
var newCurrentActors = am.ActorsInBox...
if (vRange.Length >= 1)
    newCurrentActors = newCurrentActors.Where(Math.Abs(a.CenterPosition.Z - position.Z) <= vRange.Length);
currentActors  = newCurrentActors.ToList();
```
Also maybe a `VerticalLength` and `VerticalLengthSquared` would be good to introduce? (If you do, check against the squared distance.)